db/app/models.js

```diff
@@ -21,7 +21,44 @@ function selectArticles() {
     .then((result) => result.rows);
 }
 
+function selectArticleByID(id) {
+  return db
+    .query(
+      `
+      SELECT * 
+      FROM articles 
+      WHERE article_id = $1;
+    `,
+      [id]
+    )
+    .then((result) => {
+      return result.rows.length !== 0
+        ? result.rows[0]
+        : Promise.reject({ status: 404, msg: "Not Found" });
+    })
+}
+
+function selectCommentsOfArticle(id) {
+  return db
+    .query(
+      `
+      SELECT * 
+      FROM comments 
+      WHERE article_id = $1
+      ORDER BY created_at DESC;
+    `,
+      [id]
+    )
+    .then((result) => {
+      return result.rows
+        ? result.rows
```

Comment:
I feel this might not work how you are expecting it too. By my understanding - result.rows will always be an array. If nothing ever comes back - it will be an empty array which is still a TRUTHY value.

There are going to be circumstances where an article may have no comments. So if an empty array comes back on the result.rows - we will need to do a further check if that article exists. If not - 404...if it does an empty array is fine :)